src/btree/bt_sync_obsolete.c

```diff
@@ -391,6 +407,24 @@ __checkpoint_cleanup_page_skip(
         return (0);
     }
 
+    if (__sync_obsolete_limit_reached(session)) {
+        *skipp = true;
+        return (0);
+    }
+
+    /*
+     * Read the page on disk if all the content of the page is not fully deleted and the most recent
+     * transaction is globally visible.
+     */
+    newest_ts = WT_MAX(addr.ta.newest_start_durable_ts, addr.ta.newest_stop_durable_ts);
+    if (!WT_TIME_AGGREGATE_HAS_STOP(&addr.ta) &&
+      __wt_txn_newest_visible_all(session, addr.ta.newest_txn, newest_ts)) {
```

Comment:
It seems that we call `__wt_txn_newest_visible_all` here and in `__checkpoint_cleanup_eligibility`, could it be redundant? The issue is that we may return true early from `__checkpoint_cleanup_eligibility` without calling `__wt_txn_newest_visible_all` at the end, therefore checking here could make sense.

Reply:
As discussed, will add a comment in `__checkpoint_cleanup_eligibility`